app/src/main/res/layout/activity_search.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  android:layout_width="match_parent"
+  android:layout_height="match_parent">
+
+  <android.support.design.widget.AppBarLayout
+    style="@style/Search.Appbar"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+
+    <android.support.v7.widget.Toolbar
+      android:id="@+id/search_toolbar"
+      android:theme="@style/Search.Toolbar"
+      android:layout_width="match_parent"
+      android:layout_height="?attr/actionBarSize">
+
+      <EditText
+        android:id="@+id/search_field"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@null"
+        android:hint="@string/search_label"
+        android:imeOptions="actionSearch"
+        android:maxLines="1" />
+
+    </android.support.v7.widget.Toolbar>
+  </android.support.design.widget.AppBarLayout>
+
+  <android.support.v4.widget.NestedScrollView
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    app:layout_behavior="@string/appbar_scrolling_view_behavior">
+
+    <LinearLayout
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:orientation="vertical">
+
+      <TextView
+        style="@style/Search.Body1"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:padding="16dp"
+        android:text="@string/speaker_list_title" />
+
+      <net.squanchy.search.view.SpeakersView
```

Comment:
"❌" (can do in separate PR but needs doing ASAP) this should be the only content of the scrollable area -- all other things (like section headers) should be created with different viewtypes in the adapter.

Avoid putting recyclerviews with `wrap_height` inside scrollable containers! If the recyclerview is not the one scrolling, there is no recycling possible, which means you might as well be dumping everything in the LinearLayout directly yourself.

You can/should do something along these lines:

```java
// ===== In your RecyclerView subclass =====
public void updateWith(MyItems items, ContentActionListener listener) {
    adapter.updateWith(items, listener);		// adapter is a field, initialized in onFinishInflate()

    if (getAdapter() == null) {
        super.setAdapter(adapter);
    }

    GridLayoutManager layoutManager = (GridLayoutManager) getLayoutManager();
    GridLayoutManager.SpanSizeLookup spanSizeLookup = adapter.createSpanSizeLookup(layoutManager.getSpanCount());
    layoutManager.setSpanSizeLookup(spanSizeLookup);
}

// ===== In your adapter ======
@IntDef({ViewTypeId.HEADER, ViewTypeId.SPEAKER /* and more in the future */})
@Retention(RetentionPolicy.SOURCE)
@interface ViewTypeId {

    int HEADER = 0;
    int SPEAKER = 1;
}

@Override
@ViewTypeId
public int getItemViewType(int position) {
    return itemsAdapter.getViewTypeAt(position);	// ItemsAdapter is the one that injects the headers etc
}

GridLayoutManager.SpanSizeLookup createSpanSizeLookup(int columnsCount) {
    return new GridSpanSizeLookup(itemsAdapter, columnsCount);
}

// ===== In your GridSpanSizeLookup =====
@Override
public int getSpanSize(int position) {
    if (itemsAdapter.isEmpty()) {		// itemsAdapter here is the same found in the RecyclerView adapter
        return SINGLE_COLUMN_SPAN_SIZE;
    }
    if (position < 0 || position >= itemsAdapter.getTotalItemsCount()) {
        String message = String.format(Locale.UK, "Position %d is not valid, must be [0, totalItemsCount)", position);
        throw new IndexOutOfBoundsException(message);
    }
    int itemViewType = itemsAdapter.getViewTypeAt(position);
    return getSpanSizeFor(itemViewType);
}

private int getSpanSizeFor(@ViewTypeId int viewTypeId) {
    if (viewTypeId == ViewTypeId.HEADER) {
        return columnsCount;
    } else {
        return SINGLE_COLUMN_SPAN_SIZE;
    }
}

// ===== In your ItemsAdapter =====
int getTotalItemsCount() {
    return items.getSize() + getAdditionalItemsCount();
}

private int getAdditionalItemsCount() {
    // This basically is the number of headers etc you might have
}
```

Up for helping up with this if needed, hope the code above is enough guidance but don't hesitate to ask.